### PR TITLE
[FEAT] implementation of the ORM and all inventory routes

### PR DESCRIPTION
**Pull request by RainproDash**
Here is the PR for the ORM and all the inventory routes

### CODE REVIEW

File: backend/src/app.ts [hunk 1]
```diff
@@ -12,6 +14,15 @@ dotenv.config({ path: '.env' });
 const app = express();
 const port = 3000;
```
Comment (alanschnegg) [inline backend/src/app.ts:28]: Unused variable

File: backend/src/config/db.ts [hunk 2]
```diff
@@ -1,34 +1,20 @@
 import dotenv from 'dotenv';
-import postgres from 'postgres';
+import { DataSource, In } from 'typeorm';
+import { Inventory } from '../entity/inventory.js';
 
 dotenv.config();
 
-const { PGHOST, PGDATABASE, PGUSER, PGPASSWORD, PGPORT, ENDPOINT_ID } = process.env;
+const { PGHOST, PGDATABASE, PGUSER, PGPASSWORD, PGPORT, ENDPOINT_ID, } = process.env;
```
Comment (alanschnegg) [inline backend/src/config/db.ts]: Either remove the comma or move to a column

File: backend/src/routes/inventory.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        res.status(200).json(result);
+    } catch (err) {
+        res.status(500).json({ err: `Error during data retrieval` });
+    }
+});
+
+inventoryRouter.post('/', async (req: Request, res: Response) => {
+    try {
+        await myDataSource.getRepository(Inventory).createQueryBuilder('inventory').insert().values(req.body).execute();
+        res.status(200).json({ message: `Data inserted` });
```
Comment (alanschnegg) [inline backend/src/routes/inventory.ts]: Should be a code 201
Return the created object in the response

File: backend/src/routes/inventory.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    } catch (err) {
+        res.status(500).json({ err: `Error during data insertion` });
+    }
+});
+
+inventoryRouter.delete('/:id', async (req: Request, res: Response) => {
+    try {
+        await myDataSource.getRepository(Inventory).createQueryBuilder('inventory').delete().where('inventory.id = :id', { id: req.params.id }).execute();
+        res.status(200).json({ message: `Data deleted` });
```
Comment (alanschnegg) [inline backend/src/routes/inventory.ts]: Code 202

File: backend/src/routes/inventory.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    } catch (err) {
+        res.status(500).json({ err : `Error during data deletion` });
+    }
+});
+
+inventoryRouter.patch('/:id', async (req: Request, res: Response) => {
+    try {
+        await myDataSource.getRepository(Inventory).createQueryBuilder('inventory').update().set(req.body).where('inventory.id = :id', { id: req.params.id }).execute();
+        res.status(200).json({ message: `Data updated` });
```
Comment (alanschnegg) [inline backend/src/routes/inventory.ts]: Return the updated object

**Review by alanschnegg — CHANGES_REQUESTED** [review]

File: backend/src/app.ts [hunk 6]
```diff
@@ -1,4 +1,6 @@
-import express, { Request, Response } from 'express';
+import express , {Router, Request, Response} from 'express';
+import { myDataSource } from './config/db.js';
```
Comment (thmarinho) [inline backend/src/app.ts]: rename `myDataSource` into something else

File: backend/src/routes/inventory.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+import { Inventory } from "../entity/inventory.js";
+
+const inventoryRouter = Router();
+
+inventoryRouter.get('/', async (req: Request, res: Response) => {
+    try {
+        const result = await myDataSource.getRepository(Inventory).createQueryBuilder('inventory').getMany();
+        res.status(200).json(result);
+    } catch (err) {
+        res.status(500).json({ err: `Error during data retrieval` });
```
Comment (thmarinho) [inline backend/src/routes/inventory.ts]: This message isn't clear enought

File: backend/src/routes/inventory.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    }
+});
+
+inventoryRouter.get('/:id', async (req: Request, res: Response) => {
+    try {
+        const result = await myDataSource.getRepository(Inventory).createQueryBuilder('inventory').where('inventory.id = :id', { id: req.params.id }).getOne();
+        res.status(200).json(result);
+    } catch (err) {
+        res.status(500).json({ err: `Error during data retrieval` });
```
Comment (thmarinho) [inline backend/src/routes/inventory.ts]: Same here

File: backend/src/routes/inventory.ts [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    }
+});
+
+inventoryRouter.get('/category/:id', async (req: Request, res: Response) => {
+    try {
+        const result = await myDataSource.getRepository(Inventory).createQueryBuilder('inventory').where('inventory.category = :category', { category: req.params.id }).getMany();
+        res.status(200).json(result);
+    } catch (err) {
+        res.status(500).json({ err: `Error during data retrieval` });
```
Comment (thmarinho) [inline backend/src/routes/inventory.ts]: Same here

File: backend/src/routes/inventory.ts [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    }
+});
+
+inventoryRouter.post('/', async (req: Request, res: Response) => {
+    try {
+        await myDataSource.getRepository(Inventory).createQueryBuilder('inventory').insert().values(req.body).execute();
+        res.status(200).json({ message: `Data inserted` });
+    } catch (err) {
+        res.status(500).json({ err: `Error during data insertion` });
```
Comment (thmarinho) [inline backend/src/routes/inventory.ts]: Same here

File: backend/src/routes/inventory.ts [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+    }
+});
+
+inventoryRouter.delete('/:id', async (req: Request, res: Response) => {
+    try {
+        await myDataSource.getRepository(Inventory).createQueryBuilder('inventory').delete().where('inventory.id = :id', { id: req.params.id }).execute();
+        res.status(200).json({ message: `Data deleted` });
+    } catch (err) {
+        res.status(500).json({ err : `Error during data deletion` });
```
Comment (thmarinho) [inline backend/src/routes/inventory.ts]: Same here

**Review by thmarinho — CHANGES_REQUESTED** [review]

**Review by thmarinho — APPROVED** [review]

**Review by alanschnegg — APPROVED** [review]

**Pull request merged**